listenbrainz/webserver/static/js/src/LastFMImporter.tsx

```diff
@@ -314,8 +304,21 @@ export default class LastFmImporter extends React.Component<
       );
       this.setState({ msg });
     }
+  }
+
+  async startImport() {
+    this.updateModalAction(<p>Your import from Last.fm is starting!</p>, false);
+    this.latestImportTime = await this.APIService.getLatestImport(
+      this.userName
+    );
+    this.incrementalImport = this.latestImportTime > 0;
+    this.playCount = await this.getTotalNumberOfScrobbles();
+    this.totalPages = await this.getNumberOfPages();
+    this.page = this.totalPages; // Start from the last page so that oldest scrobbles are imported first
+
+    await this.importLoop(); // import pages
```

Comment:
It would be wise to have a try-catch block around this call to `importLoop` to make sure any error in the promise won't be swallowed (.

In the catch block we have an opportunity to show the user a friendly error message letting them know the import failed (and why—by showing the error on screen).

That also means that the test you wrote ("should not contain any uncaught exceptions") won't be testing uncaught errors thrown by `importLoop` anymore, and we should instead rely on another testing method.
[you can keep the test which will catch other possible uncaught errors]

That can be for example checking instead the final message the user sees, with a test ensuring that in normal circumstances the user sees a success message, and another test to deliberately fail `importLoop` somehow and ensure the error is caught and shown to the user.

Don't hesitate to ask me questions if any part of this is unclear or if you'd like pointers on how to achieve it :)